tiledb/sm/c_api/tiledb.cc

```diff
@@ -3385,6 +3385,29 @@ int32_t tiledb_array_get_metadata_from_index(
   return TILEDB_OK;
 }
 
+int32_t tiledb_array_metadata_has_key(
+    tiledb_ctx_t* ctx,
+    tiledb_array_t* array,
+    const char* key,
+    tiledb_datatype_t* value_type,
```

Comment:
Why do we need to pass `value_type` here?

Reply:
I thought it could be useful if the user also wants to know the stored type in one call. If you think it is unnecessary then I will take it out.

Reply:
I think it is a good idea, let's keep it.